frontend/src/pages/modelServing/screens/metrics/MetricsChart.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react';
+import {
+  Card,
+  CardBody,
+  CardTitle,
+  EmptyState,
+  EmptyStateIcon,
+  Spinner,
+  Title,
+} from '@patternfly/react-core';
+import {
+  Chart,
+  ChartArea,
+  ChartAxis,
+  ChartGroup,
+  ChartThreshold,
+  ChartVoronoiContainer,
+  getResizeObserver,
+} from '@patternfly/react-charts';
+import { convertTimestamp, formatToShow, getThresholdData } from './utils';
+import { ModelServingMetricsContext } from './ModelServingMetricsContext';
+import { CubesIcon } from '@patternfly/react-icons';
+import { ContextResourceData, PrometheusQueryRangeResultValue } from '../../../../types';
+import { TimeframeTime } from '../const';
+
+type MetricsChartProps = {
+  title: string;
+  color: string;
+  metrics: ContextResourceData<PrometheusQueryRangeResultValue>;
+  unit?: string;
+  threshold?: number;
+};
+
+const MetricsChart: React.FC<MetricsChartProps> = ({ title, color, metrics, unit, threshold }) => {
+  const bodyRef = React.useRef<HTMLDivElement>(null);
+  const [chartWidth, setChartWidth] = React.useState(0);
+  const { currentTimeframe, lastUpdateTime } = React.useContext(ModelServingMetricsContext);
+
+  const processedData = React.useMemo(
+    () =>
+      metrics.data?.map((data) => ({
+        x: data[0] * 1000,
+        y: parseInt(data[1]),
+        name: title,
+      })) || [],
+    [metrics, title],
+  );
+
+  const maxValue = Math.max(...processedData.map((e) => e.y));
+
+  const hasData = processedData.length > 0;
+
+  React.useEffect(() => {
+    const ref = bodyRef.current;
+    let observer: ReturnType<typeof getResizeObserver>;
```

Comment:
oooh, you're going to get an error if it can't find the ref and then needs to unload. Either check in your return of the hook for the function or set this to `() => undefined` or something initially;